tsl/src/nodes/decompress_chunk/vector_predicates.c

```diff
@@ -114,6 +121,118 @@ vector_booleq(const ArrowArray *arrow, Datum arg, uint64 *restrict result)
 	}
 }
 
+void
+vector_uuidne(const ArrowArray *arrow, Datum arg, uint64 *restrict result)
+{
+	pg_uuid_t *uuid = DatumGetUUIDP(arg);
+	const size_t n = arrow->length;
+	const uint16 bitmap_words = (n + 63) / 64;
+	const uint64 *restrict validity = (const uint64 *) arrow->buffers[0];
+	/*
+	 * The values are treated as a pair of uint64 values.
+	 * The alignment is guaranteed by the uuid_decompress_all function.
+	 */
+	const uint64 *restrict values = (const uint64 *) arrow->buffers[1];
+	/*
+	 * No assumptions are being made about the alignment of the argument UUID,
+	 * so we copy the values to a local array.
+	 */
+	uint64 arg_values[2];
+	memcpy(arg_values, uuid->data, sizeof(arg_values));
+
+	/* Filter out null values, so we may save work for later. */
+	if (validity)
+	{
+		for (uint16 i = 0; i < bitmap_words; i++)
+		{
+			result[i] &= validity[i];
+		}
+	}
+
+	for (size_t outer = 0; outer < n / 64; outer++)
+	{
+		uint64 word = 0;
+		if (result[outer] == 0)
+			continue; /* no need to check the values if the result is false */
+
+		for (size_t inner = 0; inner < 64; inner++)
+		{
+			const bool valid = (values[outer * 64 + inner * 2] != arg_values[0] ||
+								values[outer * 64 + inner * 2 + 1] != arg_values[1]);
+			word |= ((uint64) valid) << inner;
+		}
+		result[outer] &= word;
+	}
+
+	if (n % 64 && result[n / 64 + 1] != 0)
+	{
+		uint64 tail_word = 0;
+		for (size_t i = (n / 64) * 64; i < n; i++)
+		{
+			const bool valid =
+				(values[i * 2] != arg_values[0] || values[i * 2 + 1] != arg_values[1]);
+			tail_word |= ((uint64) valid) << (i % 64);
+		}
+		result[n / 64] &= tail_word;
+	}
+}
+
+void
+vector_uuideq(const ArrowArray *arrow, Datum arg, uint64 *restrict result)
+{
+	pg_uuid_t *uuid = DatumGetUUIDP(arg);
+	const size_t n = arrow->length;
+	const uint16 bitmap_words = (n + 63) / 64;
+	const uint64 *restrict validity = (const uint64 *) arrow->buffers[0];
+	/*
+	 * The values are treated as a pair of uint64 values.
+	 * The alignment is guaranteed by the uuid_decompress_all function.
+	 */
+	const uint64 *restrict values = (const uint64 *) arrow->buffers[1];
+	/*
+	 * No assumptions are being made about the alignment of the argument UUID,
+	 * so we copy the values to a local array.
+	 */
+	uint64 arg_values[2];
+	memcpy(arg_values, uuid->data, sizeof(arg_values));
+
+	/* Filter out null values, so we may save work for later. */
+	if (validity)
+	{
+		for (uint16 i = 0; i < bitmap_words; i++)
+		{
+			result[i] &= validity[i];
+		}
+	}
+
+	for (size_t outer = 0; outer < n / 64; outer++)
+	{
+		uint64 word = 0;
+		if (result[outer] == 0)
+			continue; /* no need to check the values if the result is false */
+
+		for (size_t inner = 0; inner < 64; inner++)
+		{
+			const bool valid = (values[outer * 64 + inner * 2] == arg_values[0] &&
+								values[outer * 64 + inner * 2 + 1] == arg_values[1]);
+			word |= ((uint64) valid) << inner;
+		}
+		result[outer] &= word;
+	}
+
+	if (n % 64 && result[n / 64 + 1] != 0)
```

Comment:
not sure the check for `result[n / 64 + 1]` actually improves performance, it's a pretty marginal case, I'd drop it